File: language_construct_performance/python/scripts/run_all.py
import os
import sys
import subprocess

def run_all():
    group = "all"
    if len(sys.argv) > 1:
        group = sys.argv[1]

    print(f"Running Python benchmarks (group: {group}) sequentially in isolated processes...\n")
    
    objects_scripts = [
        'scripts/naive/plain_obj_fixed_properties.py',
        'scripts/idiomatic/plain_obj_fixed_properties.py',
        'scripts/naive/value_obj_fixed_properties.py',
        'scripts/idiomatic/value_obj_fixed_properties.py',
        'scripts/naive/plain_obj_variable_properties.py',
        'scripts/idiomatic/plain_obj_variable_properties.py',
        'scripts/naive/value_obj_variable_properties.py',
        'scripts/idiomatic/value_obj_variable_properties.py'
    ]

    json_scripts = [
        'scripts/naive/json_encoding.py',
        'scripts/idiomatic/json_encoding.py',
    ]

    if group == "objects":
        scripts = objects_scripts
    elif group == "json":
        scripts = json_scripts
    else:
        scripts = objects_scripts + json_scripts
    
    base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
    
    for script in scripts:
        print("=========================================")
        print(f"Running {script}...")
        print("=========================================")
        
        script_path = os.path.join(base_dir, script)
        result = subprocess.run([sys.executable, script_path], cwd=base_dir)
        
        if result.returncode != 0:
            print(f"Error running {script}")
            sys.exit(result.returncode)
            
        print()
        
    print("All benchmarks completed!")
    print("=========================================")
    print("Updating README.md from stats...")
    print("=========================================")
    
    update_readme_path = os.path.join(base_dir, 'scripts', 'update_readme.py')
    result = subprocess.run([sys.executable, update_readme_path], cwd=base_dir)
    
    if result.returncode != 0:
        print("Error updating README.md")
    else:
        print("README.md successfully updated!")
        
if __name__ == "__main__":
    run_all()

File: language_construct_performance/go/scripts/run_all.go
package main

import (
	"fmt"
	"os"
	"os/exec"
)

func runScript(name string) error {
	fmt.Println("=========================================")
	fmt.Printf("Running %s...\n", name)
	fmt.Println("=========================================")
	
	cmd := exec.Command("go", "run", "scripts/"+name+".go", "scripts/benchmark_utils.go")
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	
	if err := cmd.Run(); err != nil {
		fmt.Printf("Error running %s: %v\n", name, err)
		return err
	}
	fmt.Println("\n")
	return nil
}

func main() {
	group := "all"
	if len(os.Args) > 1 {
		group = os.Args[1]
	}

	fmt.Printf("Running benchmarks (group: %s) sequentially in isolated processes...\n\n", group)

	objectsScripts := []string{
		"scripts/naive/plain_obj_fixed_properties.go",
		"scripts/idiomatic/plain_obj_fixed_properties.go",
		"scripts/naive/value_obj_fixed_properties.go",
		"scripts/idiomatic/value_obj_fixed_properties.go",
		"scripts/naive/plain_obj_variable_properties.go",
		"scripts/idiomatic/plain_obj_variable_properties.go",
		"scripts/naive/value_obj_variable_properties.go",
		"scripts/idiomatic/value_obj_variable_properties.go",
	}

	jsonScripts := []string{
		"scripts/naive/json_encoding.go",
		"scripts/idiomatic/json_encoding.go",
	}

	var scripts []string
	if group == "objects" {
		scripts = objectsScripts
	} else if group == "json" {
		scripts = jsonScripts
	} else {
		scripts = append(objectsScripts, jsonScripts...)
	}

	for _, script := range scripts {
		if err := runScript(script); err != nil {
			os.Exit(1)
		}
	}

	fmt.Println("All benchmarks completed!")
	fmt.Println("=========================================")
	fmt.Println("Updating README.md from stats...")
	fmt.Println("=========================================")
	
	cmd := exec.Command("go", "run", "scripts/update_readme.go")
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	
	if err := cmd.Run(); err != nil {
		fmt.Println("Error updating README.md")
		os.Exit(1)
	} else {
		fmt.Println("README.md successfully updated!")
	}
}
